main.cpp

```diff
@@ -56,64 +56,78 @@ Create a branch named Part1
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    int value;
+    std::string name;
+    
+    T(int v, const char* name)    //1 . I'm wondering about casting from char* to std::string here? It only shows the firs character in the console
+    {
+        value = v;    //2
+        this->name = *name;    //3
+    }
 };
 
-struct <#structName1#>                                //4
+struct StructOne    //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b)    //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if(a->value < b->value) return a;
+        if(a->value > b->value) return b;
         return nullptr;
     }
```

Comment:
What do we always do before using pointers?

- [ ] apply safe pointer usage.

Reply:
Sorry, you were really clear about safe pointer usage.